x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/StringUtils.java

```diff
@@ -166,6 +167,31 @@ public static String toString(Object value, SqlVersion sqlVersion) {
             return sb.toString();
         }
 
+        // multivalue
+        if (value instanceof Collection) {
+            final StringBuilder sb = new StringBuilder();
+            sb.append('[');
+            Collection<?> values = (Collection<?>) value;
+            values.forEach(x -> {
+                // quote strings and `\`-escape the `"` character inside them
+                if (x instanceof String) { // TODO: IPs should ideally not be quoted.
+                    sb.append('"');
+                    sb.append(((String) x).replace("\"", "\\\""));
+                    sb.append('"');
+                } else if (x == null) {
+                    sb.append("NULL");
+                } else {
+                    sb.append(toString(x, sqlVersion));
+                }
+                sb.append(',');
+            });
+            if (sb.length() > 1) { // strip trailing comma
+                sb.deleteCharAt(sb.length() - 1);
+            }
+            sb.append(']');
+            return sb.toString();
+        }
```

Comment:
Better to use `StringJoiner` since it's part of the JDK and handles the exact case of prefix/suffix plus delimiter.

Reply:
Nice, thanks, applied.